pkg/naming/names.go

```diff
@@ -101,11 +102,16 @@ func IndexFromName(n string) (int32, error) {
 
 // ImageToVersion strips version part from container image.
 func ImageToVersion(image string) (string, error) {
-	parts := strings.Split(image, ":")
-	if len(parts) != 2 || len(parts[1]) == 0 {
-		return "", errors.New(fmt.Sprintf("Invalid image name: %s", image))
+	named, err := reference.ParseNormalizedNamed(image)
+	if err != nil {
+		return "", fmt.Errorf("can't parse image: %w", err)
+	}
+	tagged, ok := named.(reference.NamedTagged)
+	if !ok {
+		return "", fmt.Errorf("invalid, non-tagged image reference: %s", image)
```

Comment:
nit: when a type doesn't implement an interface, it might be useful to enrich the message with the type that failed as well

Reply:
done